src/commands/org/create/sandbox.ts

```diff
@@ -79,18 +81,25 @@ export default class CreateSandbox extends SandboxCommandBase<SandboxCommandResp
         return Promise.resolve(name);
       },
     }),
-    clone: Flags.string({
+    'source-sandbox-name': Flags.string({
       char: 'c',
-      summary: messages.getMessage('flags.clone.summary'),
-      description: messages.getMessage('flags.clone.description'),
+      summary: messages.getMessage('flags.source-sandbox-name.summary'),
+      description: messages.getMessage('flags.source-sandbox-name.description'),
+      exclusive: ['license-type', 'source-id'],
+      deprecateAliases: true,
+      aliases: ['clone'],
+    }),
+    'source-id': Flags.string({
```

Comment:
this can be a `salesforceId` flag, example:
https://github.com/salesforcecli/plugin-org/blob/0e7d3a7f6ae98240404311fc30a1ed4df3a32eb9/src/commands/org/create/scratch.ts#L148-L155

I don't see the API mentions sandbox IDs start with a particular prefix so we can omit the `startWith` option.

`Flags.salesforceId` will validate that the ID length is 15 or 18.